moveit_core/robot_model/include/moveit/robot_model/robot_model.h

```diff
@@ -239,6 +239,19 @@ class RobotModel
   /** \brief Get a link by its name. Output error and return NULL when the link is missing. */
   LinkModel* getLinkModel(const std::string& link);
 
+  /** \brief Get the latest link upwards the kinematic tree, which is only connected via fixed joints
+   *
+   * This is useful, if the link should be warped to a specific pose using updateStateWithLinkAt().
+   * As updateStateWithLinkAt() warps only the specified link and its descendants, you might not
+   * achieve what you expect, if link is an abstract frame name. Considering the following example:
+   * root -> arm0 -> ... -> armN -> wrist -- grasp_frame
+   *                                      -- palm -> end effector ...
+   * Calling updateStateWithLinkAt(grasp_frame), will not warp the end effector, which is probably
+   * what you went for. Instead, updateStateWithLinkAt(getRigidlyConnectedParentLinkModel(grasp_frame), ...)
+   * will actually warp wrist (and all its descendants).
+   */
+  static const moveit::core::LinkModel* getRigidlyConnectedParentLinkModel(const LinkModel* link);
```

Comment:
My main question with this function is where to put it.

1. Leave it here in `RobotModel` as a static function?
2. Define it as a member of `LinkModel`?
3. Put it (as a static function again) into `RobotState`. i.e. the vicinity of `updateStateWithLinkAt()`?

The order resembles my preference. @davetcoleman @v4hn Any other opinions?

Reply:
Personally I think you picked the best spot already. I vote to keep it in `RobotModel` together with all the `getLink` functions.

Reply:
@davetcoleman If you agree too, I think this is ready for merging.

Reply:
+1 LinkModel. Def not RobotState but option 2 works also for me.

Reply:
So then we agree on this. Could anybody of you finally accept / merge this PR?